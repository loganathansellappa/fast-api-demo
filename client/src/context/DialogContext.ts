import { Dispatch } from 'react';
import { createContext, SetStateAction, useContext } from 'react';

export type DialogState = {
  isOpen: boolean;
  content: React.ReactNode | null;
  handler: () => void;
  title: string;
  noBtnText: string;
  yesBtnText: string;
};

export type DialogContextType = {
  dialog: DialogState;
  setDialog: Dispatch<SetStateAction<DialogState>>;
};

export const DialogContext = createContext<DialogContextType | null>(null);


export const useDialogContext = () => {
  /* eslint-disable  @typescript-eslint/no-explicit-any */
  const context: any = useContext(DialogContext)
  if (context === undefined) {
    throw new Error('useAxiosClient must be used within a AxiosClientProvider')
  }
  return context;
}