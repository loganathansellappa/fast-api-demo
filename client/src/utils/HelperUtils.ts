import { toast } from 'react-toastify';

export type ServerConfig = {
    baseUrl: string;
};
export const serverData = () => {
    return {
        baseUrl: process.env.DATA_URL,
    } as ServerConfig;
};

export enum MissionState {
    PreFlight = 'Pre-flight',
    Flight = 'In-Flight',
    PostFlight = 'Post-Flight'
}


export enum MutationType {
    PUT = 'PUT',
    POST = 'POST',
    DELETE = 'DELETE'
}

export const showSuccessToast = (message: string, id: string | number = 1) => {
    toast.dismiss();
    toast.success(message, { toastId: id, autoClose: 500, style: { background: '#5CDD00', color: "white" } });
};

export const showErrorToast = (message: string, id: string | number) => {
    toast.dismiss();
    toast.error(message, { toastId: id, autoClose: 500 });
};
