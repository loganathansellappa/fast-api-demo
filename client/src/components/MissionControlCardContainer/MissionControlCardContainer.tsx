import { MissionControlCard } from '../MissionControlCard/MissionControlCard.tsx';
import { MissionState } from '../../utils/HelperUtils.ts';
import { MissionControlSchema } from '../../@types/zodSchema.ts';
import { useDialogContext } from '../../context/DialogContext.ts';
import { useCallback } from 'react';
import { DeleteMissionControl } from '../DeleteMissionControl/DeleteMissionControl.tsx';

export type MissionControlCardContainerProps = {
  missionControl: MissionControlSchema[];
  title: MissionState;
}

export const MissionControlCardContainer = ({ missionControl, title }: MissionControlCardContainerProps) => {

  const { setDialog } = useDialogContext();

  const closeDialog = useCallback(() => {
    setDialog({
      isOpen: false,
      text: "",
      handler: null,
      noBtnText: "",
      yesBtnText:""
    });
  }, [setDialog]);

  const openDialog = useCallback((id: number) => {
    setDialog({
      isOpen: true,
      title: 'DELETE - Mission',
      content: <DeleteMissionControl id={id} onCancel={closeDialog}  />
    });
  }, [setDialog, closeDialog]);

  const borderColor =
    title === MissionState.PreFlight ? 'orange' : title === MissionState.PostFlight ? 'green' : 'blue'

  return (
    <>
      <div className="bg-gray-100 text-black font-bold mb-12 rounded sticky">{title}</div>
      <div className="no-scrollbar overflow-y-scroll ">
        {missionControl.map((control: MissionControlSchema) => (
          <MissionControlCard key={control.id} missionControl={control} borderColor={borderColor}
                              onDeleteClick={openDialog} />
        ))}
      </div>
    </>
  );
}
