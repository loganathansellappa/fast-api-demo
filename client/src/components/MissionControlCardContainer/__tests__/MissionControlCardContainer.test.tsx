import { render } from "@testing-library/react";
import { MissionControlCardContainer, MissionControlCardContainerProps } from '../MissionControlCardContainer.tsx';
import { createContext } from 'react';
import * as Contexts from '../../../context/DialogContext.ts';
import { MissionState } from '../../../utils/HelperUtils.ts';

let element: { container: HTMLElement };

beforeEach(() => {
  const context = createContext<Contexts.DialogContextType>({
    dialog: {
      isOpen: false,
      content: null,
      handler: () => jest.fn(),
      title: "",
      noBtnText: "",
      yesBtnText: "",
    },
    setDialog: () => {}
  })
  jest
    .spyOn(Contexts, "useDialogContext")
    .mockReturnValue(context);
  const props: MissionControlCardContainerProps = {
    missionControl:  [
        {
          "title": "string",
          "description": "string",
          "mission_state": "Pre-flight",
          "id": 319,
          "created_at": "2024-02-14T00:45:50",
          "updated_at": "2024-02-14T00:54:26"
      }],
    title: MissionState.PreFlight,
  };
  element = render(<MissionControlCardContainer {...props} />);
});

describe("MissionControlCardContainer", () => {

  test("renders MissionControlCardContainer component", () => {
    expect(element.container.firstChild).toMatchSnapshot();
  });

  test("displays title", () => {
    expect(element.container.firstChild?.textContent).toContain(MissionState.PreFlight);
  });
});
