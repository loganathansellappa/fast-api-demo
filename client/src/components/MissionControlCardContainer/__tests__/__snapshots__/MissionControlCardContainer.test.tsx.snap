// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionControlCardContainer renders MissionControlCardContainer component 1`] = `
<div
  class="bg-gray-100 text-black font-bold mb-12 rounded sticky"
>
  Pre-flight
</div>
`;
